src/Costumes.cpp

```diff
@@ -117,6 +119,16 @@ bool Costumes::CanUseItem(Player *player, ItemTemplate const *item, InventoryRes
         return true;
     }
 
+    if (!canUseInBg && player->InBattleground())
+    {
+        return false;
+    }
+
+    if (!canUseInArena && player->InArena())
+    {
+        return false;
+    }
+
     if (!canUseInCombat && player->IsInCombat())
     {
         result = InventoryResult::EQUIP_ERR_CANT_DO_RIGHT_NOW;
```

Comment:
This is not from your PR but I just realized it would be a better fit, while we're at it 😄
```suggestion
        result = InventoryResult::EQUIP_ERR_NOT_IN_COMBAT;
```